crates/matrix-sdk-base/src/client.rs

```diff
@@ -872,21 +937,21 @@ impl BaseClient {
                     }
 
                     ambiguity_cache.handle_event(&changes, room_id, &member).await?;
+                }
```

Comment:
So I'm still not sure what to do about this. One idea is that there should be a way to know which users are currently being added via the sync so that we can skip them, I think Hydrogen does something along those lines.

Now how to progress on this, I think this is a sufficient edge case so we can leave it for later. Could you please open an issue and add a TODO here explaining the problem briefly and point it to the issue.

One thing that is problematic with this code and should be fixed before the merge is that we don't mark users to be tracked if they have a stale leave membership in the store and we now get a joined membership for them.

The same is true for the ambiguity cache, I think the whole `if self.store.get_member_event(..).await?.is_none() {` needs to be gone, no?

Reply:
I don't fully get what you mean by this:
> One thing that is problematic with this code and should be fixed before the merge is that we don't mark users to be tracked if they have a stale leave membership in the store and we now get a joined membership for them.

But I have remove the whole if, as I agree.

I have also wrote a comment and opened an issue.

Reply:
> I don't fully get what you mean by this

Suppose we were in a room where Bob had a membership event in the `leave`, we leave the room and rejoin. While we weren't part of the room Bob has joined as well.

Now we won't insert Bob into the `user_ids` `BTreeSet` since `self.store.get_member_event(..).await?.is_none()` will be `false`. This in turn means that Bob won't be marked as tracked using the `o.update_tracked_users(user_ids.iter().map(Deref::deref)).await` call.

This has further implications since if a user is tracked or not decides if we're going to ask the server for all the devices a user has. Finally Bob's devices will be missing from our store. Once we try to send a message to such a room, Bob won't receive the decryption keys and won't be able to decrypt the message.

> But I have removed the whole if, as I agree.

Yes that's all that was needed, thanks.